File: RPi Experiments/RPi2-1/C/11_rgbLed.c
#include <wiringPi.h>
#include <softPwm.h>
#include <stdio.h>

#define uchar unsigned char

#define redPin 0 //the number of the LED module red pin
#define greenPin 1 //the number of the LED module green pin
#define bluePin 2 //the number of the LED module blue pin

void ledInit(void)
{
	softPwmCreate(redPin, 0, 100); //initialize the LED red pin as pwm output
	softPwmCreate(greenPin, 0, 100); //initialize the LED green pin as pwm output
	softPwmCreate(bluePin, 0, 100); //initialize the LED blue pin as pwm output
}

//Color generating function 
void ledColorSet(uchar red, uchar green, uchar blue)
{
	softPwmWrite(redPin, red); //set the brightness of LED red  
	softPwmWrite(greenPin, green); //set the brightness of LED green 
	softPwmWrite(bluePin, blue); //set the brightness of LED blue 
}

int main(void)
{
    int i;
    int ms;

    if(wiringPiSetup() == -1)
    { 
		printf("setup wiringPi failed !"); //when initialize wiring failed,print messageto screen
		return 1; 
	}
	
	ledInit();

    while(1)
    {
        ms = 1000;
        //Basic colors:
		ledColorSet(0xff,0x00,0x00); //turn the RGB LED red	
		delay(ms);
		ledColorSet(0x00,0xff,0x00); //turn the RGB LED green
		delay(ms);
		ledColorSet(0x00,0x00,0xff); //turn the RGB LED blue
		delay(ms);
        //Blended colors:
		ledColorSet(0xed,0x6d,0x00); //turn the RGB LED orange
		delay(ms);
		ledColorSet(0xff,0xd7,0x00); //turn the RGB LED yellow
		delay(ms);
		ledColorSet(0x00,0x2e,0x5a); //turn the RGB LED indigo
		delay(ms);
		ledColorSet(0x80,0x00,0x80); //turn the RGB LED purple
		delay(ms);
	}

	return 0;
}

File: RPi Experiments/RPi2-1/Python/11_rgbLed.py
#!/usr/bin/env python
import RPi.GPIO as GPIO
import time

redPin = 11 # physical pin number for the LED red
greePin = 12 # physical pin number for the LED green
bluePin = 13 # physical pin number for the LED blue

def setup():
    global pR
    global pG
    global pB

    GPIO.setwarnings(False) # don't show warnings if GPIO is in use
    GPIO.setmode(GPIO.BOARD) # use GPIO by physical location

    GPIO.setup(redPin, GPIO.OUT) # initialize the LED red pin as output
    GPIO.setup(greePin, GPIO.OUT) # initialize the LED green pin as output
    GPIO.setup(bluePin, GPIO.OUT) # initialize the LED blue pin as output

    GPIO.output(redPin, GPIO.HIGH) # turn off LED red
    GPIO.output(greePin, GPIO.HIGH) # turn off LED green
    GPIO.output(bluePin, GPIO.HIGH) # turn off LED blue

    pR = GPIO.PWM(redPin, 100) # setup the LED red pin for PWM, 100 Hz is enough not to see a blinking
    pG = GPIO.PWM(greePin, 100) # setup the LED green  pin for PWM, 100 Hz is enough not to see a blinking
    pB = GPIO.PWM(bluePin, 100) # setup the LED blue pin for PWM, 100 Hz is enough not to see a blinking

    pR.start(0) # start the PWM instance with 0 duty cycle
    pG.start(0) # start the PWM instance with 0 duty cycle
    pB.start(0) # start the PWM instance with 0 duty cycle

# Color generating function 
def ledColorSet(red, green, blue):  
	pR.ChangeDutyCycle(red) # set the brightness of LED red   
	pG.ChangeDutyCycle(green) # set the brightness of LED green
	pB.ChangeDutyCycle(blue) # set the brightness of LED blue

def loop():
    s = 1
    while True:
        # Basic colors:
        ledColorSet(100, 0, 0) # turn the RGB LED red
        time.sleep(s)
        ledColorSet(0, 100, 0) # turn the RGB LED green
        time.sleep(s)
        ledColorSet(0, 0, 100) # turn the RGB LED blue
        time.sleep(s)
        #Blended colors:
        ledColorSet(95, 52, 0) # turn the RGB LED orange
        time.sleep(s)
        ledColorSet(100, 90, 0) # turn the RGB LED yellow
        time.sleep(s)
        ledColorSet(0, 18, 40) # turn the RGB LED indigo
        time.sleep(s)
        ledColorSet(50, 0, 50) # turn the RGB LED purple
        time.sleep(s)
		
def destroy():
	pR.stop() # stop the PWM instance
	pG.stop() # stop the PWM instance
	pB.stop() # stop the PWM instance
	GPIO.output(redPin, GPIO.HIGH) # turn off LED red
	GPIO.output(greePin, GPIO.HIGH) # turn off LED green
	GPIO.output(bluePin, GPIO.HIGH) # turn off LED blue
	GPIO.cleanup() # release resource

if __name__ == '__main__': # program start from here
	try:
		setup()
		loop()

	except KeyboardInterrupt: # when 'Ctrl+C' is pressed, the child program destroy() will be  executed.
		destroy()